src/mds/SimpleLock.cc

```diff
@@ -57,6 +57,7 @@ int SimpleLock::get_wait_shift() const {
     case CEPH_LOCK_INEST:    return 8 + 9*SimpleLock::WAIT_BITS;
     case CEPH_LOCK_IFLOCK:   return 8 +10*SimpleLock::WAIT_BITS;
     case CEPH_LOCK_IPOLICY:  return 8 +11*SimpleLock::WAIT_BITS;
+    case CEPH_LOCK_IQUIESCE: return 8 +12*SimpleLock::WAIT_BITS;
```

Comment:
Have a problem here. 8 + 12*4 = 56; that's the shift, so given the 4 wait bits we get to 60, but that gets us into the wait flags

```
  // -- wait --
  const static uint64_t WAIT_ORDERED	 = (1ull<<61);
  const static uint64_t WAIT_SINGLEAUTH  = (1ull<<60);
  const static uint64_t WAIT_UNFREEZE    = (1ull<<59); // pka AUTHPINNABLE
```

Even if we consider that bit 4, `WAIT_REMOTEXLOCK`, is unused for quiesce, the `WAIT_STABLE << iquiesce shift` already overlaps with `WAIT_UNFREEZE`.


First, I think we should add a static assert here so that people don't return shifts that aren't compatible with the 64 bit target mask

Second, we should probably revise how that mask is used in the wait list to be able to add more locks in the future - and now

Reply:
An additional comment here is that the shift defines the waiter priority. What do you think, maybe we want the quiesce lock waiters to be processed before others, i.e. we may want to insert this with a lower shift?

Reply:
The investigation showed that after #8965 this method became redundant, simply polluting the wait bitmask. I've opened #55668 to fix that.

Reply:
I think we can close this?